File: 2015/05/2.rb
nice = 0

File.open('input.txt').each do |line|
  rule_1, rule_2 = false, false

  line.chars.each_cons(2) do |chars|
    rule_1 = true and break if line.sub(chars.join, '  ').match?(chars.join)
  end

  line.chars.each_cons(3) do |a, b, c|
    rule_2 = true and break if a == c
  end

  nice += 1 if rule_1 && rule_2
end

puts nice

File: 2015/05/2.py
import re

nice_count: int = 0

def rule_1(line: str) -> bool:
    return re.search(r'(..).*\1', line) is not None

def rule_2(line: str) -> bool:
    return re.search(r'(.).\1', line) is not None

with open("input.txt") as f:
    for line in f:
        line: str = line.strip()

        if rule_1(line) and rule_2(line):
            nice_count += 1

print(nice_count)
